Source/Core/Common/Arm64Emitter.h

```diff
@@ -772,10 +789,15 @@ class ARM64XEmitter
 
   // Logical (immediate)
   void AND(ARM64Reg Rd, ARM64Reg Rn, u32 immr, u32 imms, bool invert = false);
+  void AND(ARM64Reg Rd, ARM64Reg Rn, LogicalImm imm);
   void ANDS(ARM64Reg Rd, ARM64Reg Rn, u32 immr, u32 imms, bool invert = false);
+  void ANDS(ARM64Reg Rd, ARM64Reg Rn, LogicalImm imm);
   void EOR(ARM64Reg Rd, ARM64Reg Rn, u32 immr, u32 imms, bool invert = false);
+  void EOR(ARM64Reg Rd, ARM64Reg Rn, LogicalImm imm);
   void ORR(ARM64Reg Rd, ARM64Reg Rn, u32 immr, u32 imms, bool invert = false);
+  void ORR(ARM64Reg Rd, ARM64Reg Rn, LogicalImm imm);
   void TST(ARM64Reg Rn, u32 immr, u32 imms, bool invert = false);
```

Comment:
Is there a point in keeping the old versions? I assume we never want to generate this functions with immr/imms, and even if we want so, we can use the LogicalImm constructor.

Reply:
I was planning to clean up the calls to the immr/imms version in a follow-up PR. I could remove it in that PR.

Reply:
Fine. No need to make this one larger :)